### ISSUE TITLE
Resources Refreshing Resources

### ISSUE THREAD

**Issue by jtopjian** [hashicorp/terraform#4118]
#2011 has a need for one resource to refresh another resource so the computed attributes are grabbed.

At a high level, here's what's happening:
1. A Floating IP resource is created. It has a computed resource of the private IP of the instance it is attached to, but it doesn't actually know of this IP until the instance has been fully created. Therefore, it's unknown until the resource is refreshed.
2. The instance is created and the private IP is know available.
3. A DNS record resource is created and references the private IP via the Floating IP resource. 

Since the Floating IP resource was never refreshed, the private IP is not known. In order for this to work, there needs to be a refresh between steps 2 and 3.

In the particular scenario of #2011, there are a few ways to get around this issue (alternative ways of obtaining the private IP). But per the conversation in #2011, this functionality doesn't exist and might be applicable to other areas.

**Comment by apparentlymart** [comment]
Sorry for the long silence here!

This seems to be covering the same general problem as #2585. I closed that today not because it's been solved but because, with the benefit of a few more years of experience, we've found that this situation is relatively rare and so we think better to rely on conventions and careful documentation in providers for it, rather than adding a complex new feature to Terraform Core.
